### ISSUE TITLE
Compilation error when cross compiling for arm-unknown-linux-*

### ISSUE THREAD

**Issue by ericonr** [ftilde/unsegen_terminal#1]
Hi there! I am building the latest release of `ugdb` for Void Linux. When compiling this project for the `arm-unknown-linux-musleabihf` Rust target, I got the following error:

```
   Compiling unsegen_terminal v0.2.0
   Compiling unsegen_signals v0.2.0
error[E0308]: mismatched types
   --> /host/cargo/registry/src/github.com-1ecc6299db9ec823/unsegen_terminal-0.2.0/src/pty/mod.rs:120:65
    |
120 |         let pts_name_cstr = unsafe { ::std::ffi::CStr::from_ptr(pts_name) };
    |                                                                 ^^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*mut i8`

   Compiling unsegen_jsonviewer v0.2.0
error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: could not compile `unsegen_terminal`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

This same error doesn't seem to happen for the `i686-unknown-linux-musl` Rust target, but it seems to also happen for the `arm-unknown-linux-gnueabihf` target. So the issue is likely the platform and not the libc. Do you think this is an issue specifically in your code or something that comes from the Rust std lib?

**Comment by q66** [comment]
looks to me like a char signedness issue, `pts_name` is signed like is expected on x86, but on ARM, ppc and so on char is unsigned

**Comment by ericonr** [comment]
This

```
    pub fn ptsname(fd: libc::c_int) -> *mut libc::c_schar;
```

Should be using `libc::c_char`, which is the platform's native char.

**Comment by ftilde** [comment]
Should be fixed in fd7aa9db907dbc60dcfceea64ea82be0b9be8732 as you suggested (thanks again!) (in which I accidentally supplied the PR number instead of that of this issue.)

**Comment by ftilde** [comment]
@ericonr could you report back if you were able to successfully build ugdb for `arm-unknown-linux-gnueabihf`? If that's the case I will update its dependencies and publish a new version for it as well.

**Comment by ericonr** [comment]
@ftilde it did work, thanks! cargo automatically used version `0.2.1`, since you only specified `0.2`, so I don't know if you need to publish a new version of `ugdb` just for this.

**Comment by ericonr** [comment]
Although, if you're going to cut a release, would you mind either updating the `libgit2` dependency or having some way of disabling it? SInce we're building from the tarball, it does't even get used. But having a newer a version is a bit better, at least, since then it knows to use the system's `libgit2`.

**Comment by ftilde** [comment]
On 13.07.2020 17:51, Érico Nogueira Rolim wrote:
>@ftilde it did work, thanks! cargo automatically used version `0.2.1`, since you only specified `0.2`, so I don't know if you need to publish a new version of `ugdb` just for this.
>

Afaik, one is supposed to build packages with the `--locked` flag, which 
causes cargo to use the exact dependency versions specified in the 
`Cargo.lock` file (which for this exact reason should be checked in for 
binaries, but not for libs) to at least somewhat facilitate 
reproducibility. I think this also happens automatically if you install 
binaries via cargo.

I've just updated the dependencies of ugdb, though, so that shouldn't be 
an issue.

**Comment by ftilde** [comment]
On 13.07.2020 17:53, Érico Nogueira Rolim wrote:
>Although, if you're going to cut a release, would you mind either updating the `libgit2` dependency or having some way of disabling it? SInce we're building from the tarball, it does't even get used. But having a newer a version is a bit better, at least, since then it knows to use the system's `libgit2`.
>

I've removed the libgit2 dependency now. Does this work for you? Let me 
know if there is anything else hindering packaging. I'm happy to see 
that you want to package ugdb. :)

**Comment by ericonr** [comment]
> Afaik, one is supposed to build packages with the `--locked` flag

We use the `--locked` flag for installing, but building without it allows us to pick up library versions with bug fixes and other issues.

> I've removed the libgit2 dependency now. Does this work for you?

Certainly, this works great! I don't think there's anything else missing, at least not that I could find.

Thanks a lot! I really like using ugdb :)

**Comment by ericonr** [comment]
Please ping me once you make the new ugdb release c:

**Comment by ftilde** [comment]
On 14.07.2020 17:10, Érico Nogueira Rolim wrote:
>Please ping me once you make the new ugdb release c:

FYI: I've just published version 0.1.8.

**Comment by ericonr** [comment]
Thanks! :D
